File: go/dsa/algorithms/strings/is-subsequence.go
func isSubsequence(s string, t string) bool {
    i, j := 0, 0
    for i < len(s) && j < len(t) {
        // 1️⃣ Compare: Do characters match?
        if s[i] == t[j] {
            // 2️⃣ Match! Consume from subsequence
            i++
        }
        // 3️⃣ Always advance through main string
        j++
    }
    return i == len(s)
}

File: javascript/dsa/algorithms/strings/is-subsequence.js
function isSubsequence(s, t) {
    let i = 0, j = 0;
    while (i < s.length && j < t.length) {
        // 1️⃣ Compare: Do characters match?
        if (s[i] === t[j]) {
            // 2️⃣ Match! Consume from subsequence
            i++;
        }
        // 3️⃣ Always advance through main string
        j++;
    }
    return i === s.length;
}